code/modules/networks/computer3/computer3.dm

```diff
@@ -337,294 +337,108 @@
-			return;
-		}
-
-		function con_clear()
-		{
-			printing.length = 0;
-			objDiv.innerHTML = "";
-		}
-
-		</script>"}
-
-		dat += {"<td><a href='byond://?src=\ref[src];restart=1'>Restart</a></td>
-		</body>"}
-
-		user.Browse(dat,"window=comp3;size=455x405")
-		onclose(user,"comp3")
-	return
-
-/obj/machinery/computer3/proc/update_peripheral_menu(mob/user as mob)
-	var/count = 0
-	for (var/obj/item/peripheral/pCard in src.peripherals)
-		if (pCard.setup_has_badge)
-			user <<  output(url_encode(pCard.return_badge()),"comp3.browser:setBadge[count]")
-			count++
-
-	return
-
-/obj/machinery/computer3/Topic(href, href_list)
-	if(..())
-		return
-
-	src.add_dialog(usr)
-
-	if((href_list["command"]) && src.active_program)
-		usr << output(null, "comp3.browser:input_clear")
-		src.active_program.input_text(href_list["command"])
-		playsound(src.loc, "keyboard", 50, 1, -15)
-
-	else if(href_list["disk"])
-		if (src.diskette)
-			//Ai/cyborgs cannot press a physical button from a room away.
-			if((issilicon(usr) || isAI(usr)) && BOUNDS_DIST(src, usr) > 0)
-				boutput(usr, "<span class='alert'>You cannot press the ejection button.</span>")
-				return
-
-			for(var/datum/computer/file/terminal_program/P in src.processing_programs)
-				P.disk_ejected(src.diskette)
-
-			usr.put_in_hand_or_eject(src.diskette) // try to eject it into the users hand, if we can
-			src.diskette = null
-			usr << output(url_encode("Disk: <a href='byond://?src=\ref[src];disk=1'>-----</a>"),"comp3.browser:setInternalDisk")
-		else
-			var/obj/item/I = usr.equipped()
-			if (istype(I, /obj/item/disk/data/floppy))
-				usr.drop_item()
-				I.set_loc(src)
-				src.diskette = I
-				usr << output(url_encode("Disk: <a href='byond://?src=\ref[src];disk=1'>Eject</a>"),"comp3.browser:setInternalDisk")
-			else if (istype(I, /obj/item/magtractor))
-				var/obj/item/magtractor/mag = I
-				if (istype(mag.holding, /obj/item/disk/data/floppy))
-					I = mag.holding
-					mag.dropItem(0)
-					I.set_loc(src)
+/obj/machinery/computer3/ui_interact(mob/user, datum/tgui/ui)
+  ui = tgui_process.try_update_ui(user, src, ui)
+  if(!ui)
+    ui = new(user, src, "Terminal")
+    ui.open()
+
+
+/obj/machinery/computer3/ui_data(mob/user)
+ . = list(
+	"temp" = src.temp, // display data
+	"TermActive" = src.active_program, // is the terminal running or restarting
+	"fdisk" = src.diskette, // for showing if the internal diskette slot is filled
+	"windowName" = src.name,
+	"user" = user,
+	"fontColor" = src.setup_font_color, // display monochrome values
+	"bgColor" = src.setup_bg_color
+  )
+	if(src.setup_has_internal_disk) // the magic internal floppy drive is in here
+		. += list("peripherals" = list(list(
+		"internalFdisk" = src.setup_has_internal_disk,
+		"icon" = "save",
+		"card" = "internal",
+		"color" = src.diskette,
+		"label" = "Disk"
+		)))
+	for(var/obj/item/peripheral/periph in src.peripherals) // originally i had all this stuff in static data, but the buttons didnt update.
```

Comment:
The idea of static data is to use `update_static_data` to force an update whenever the data changes

Reply:
did that